src/System.ComponentModel.TypeConverter/src/System/ComponentModel/LicenseManager.LicenseInteropHelper.cs

```diff
@@ -47,10 +46,19 @@ private class LicenseInteropHelper
             private static object AllocateAndValidateLicense(RuntimeTypeHandle rth, IntPtr bstrKey, int fDesignTime)
             {
                 Type type = Type.GetTypeFromHandle(rth);
-                CLRLicenseContext licensecontext = new CLRLicenseContext(fDesignTime != 0 ? LicenseUsageMode.Designtime : LicenseUsageMode.Runtime, type);
-                if (fDesignTime == 0 && bstrKey != (IntPtr)0)
+                string key = null;
+                if (bstrKey != IntPtr.Zero)
                 {
-                    licensecontext.SetSavedLicenseKey(type, Marshal.PtrToStringBSTR(bstrKey));
+                    key = Marshal.PtrToStringBSTR(bstrKey);
+                }
+                return AllocateAndValidateLicense2(type, key, (fDesignTime == 1 ? true : false));
+            }
+            private static object AllocateAndValidateLicense2(Type type, string key, bool isDesignTime)
```

Comment:
I assume that you will delete the existing methods with weird signatures. Maybe add a comment what is expected to be deleted?

It would be nice if the methods that we keep did not have the `2` suffix.

Reply:
Yes. That is the plan. I can add a comment as suggested, but my larger goal is to remove this file entirely. It is entirely related to COM activation and as such should be moved into the COM activation class within SPCL. This is apart of the mop-up work once the scenario is checked in.

Reply:
Comment added.